### PR TITLE
[ios][android] 📚 Update react-native-screens to 2.0.0

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Part of #7006 

Huge shoutout to @kmagiera for fixing all discovered bugs while being under pressure of Expo SDK37 release 👏 🗣 

# How

- I've run `et update-vendored-module -m react-native-screens`.
- Added `ScreenStack` example to `native-component-list`.
- Updated dependencies of `expoview` required by `react-native-screens` (especially `androidx.fragment:fragment` needs to be at least `1.2.0`).

# Test Plan

- Screens example in `native-component-list` works as expected on both iOS and Android.
- This version was also tested in a few apps that we're working on at @software-mansion.

### CODE REVIEW

**Comment by Hirbod** [comment]
https://github.com/software-mansion/react-native-screens/pull/345

Will this be included before merge? Otherwise Expo Users, which rely on nativeStack will be lost since react navigation v5 removed it and transferred it to screens.

**Comment by Hirbod** [comment]
Native stack got merged. So hopefully you go for beta.14. Sorry for spamming on this topic

**Review by bbarthec — APPROVED** [review]

File: tools/expotools/src/commands/UpdateVendoredModule.ts [hunk 1]
```diff
@@ -659,7 +660,9 @@ async function action(options: ActionOptions) {
     };
 
     if (moduleConfig.installableInManagedApps) {
-      const versionRange = `${moduleConfig.semverPrefix || ''}${version}`;
+      const semverPrefix = (options.semverPrefix != null ? options.semverPrefix : moduleConfig.semverPrefix) || '';
```
[the remark this answers is not in the document]
Comment (tsapeta) [reply tools/expotools/src/commands/UpdateVendoredModule.ts]: This doesn't work with the TypeScript version that we already have, I'd prefer to upgrade it separately 😉

**Comment by Hirbod** [comment]
Back gesture is broken on iOS in native-screens v2
https://github.com/software-mansion/react-native-screens/pull/389/files

https://github.com/software-mansion/react-native-screens/issues/385

**Pull request merged**